src/libraries/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs

```diff
@@ -43,13 +43,19 @@ public async Task ServerAsyncAuthenticate_EachSupportedProtocol_Success(SslProto
             await ServerAsyncSslHelper(protocol, protocol);
         }
 
-        [Theory]
+        [ConditionalTheory]
         [MemberData(nameof(ProtocolMismatchData))]
         public async Task ServerAsyncAuthenticate_MismatchProtocols_Fails(
             SslProtocols serverProtocol,
             SslProtocols clientProtocol,
             Type expectedException)
         {
+
+            if ((serverProtocol & SslProtocolSupport.SupportedSslProtocols) == 0)
+            {
+                throw new SkipTestException($"None of '{serverProtocol}' requested versions is available");
```

Comment:
I don't mind the check, but this shouldn't happen should it? Based on the checks in `ProtocolMismatchData`. I'm just trying to make sure I understand this.

Reply:
There are two parts. The one side is guarded but the other is not. So we would not start server on unsupported version but the client can be anything.

We could possibly change that as well. We have tests where the client part is set of allowed protocols and the test would work as far as _any_ of them is available. It is also somewhat more complicated as the protocol may not be supported by the SSL stack but on this case the server does support but it is disabled in registry. And when it does, some of the call fail with WIn32Exception. There is also some variations I run into on Linux: The protocols may not be disabled explicitly but all the ciphers suites used by it may - as deemed weak. In that case the  API calls succeed but then the negotiation  fails with protocol mismatch. 

Perhaps we should construct this automatically e.g. create disjoined sets from all supported protocols.